apps/team-awesome-web/pages/index.tsx

```diff
@@ -1,72 +1,74 @@
+import { PlusIcon } from "../src/components/button/Button.styled";
+import { StoreContext } from "../src/components/store-context";
 import { SubHeader } from "../src/components/sub-header";
 import { TimeEntries } from "../src/components/time-entries";
 import { TimeEntryForm } from "../src/components/Form/time-entry-form";
-import { deleteTimeEntry } from "../src/services/time-entries/deleteTimeEntries";
 import * as Types from "../src/types";
-import { getTimeEntries } from "../src/services/time-entries/getTimeEntries";
-import { StoreContext } from "../src/components/store-context";
 
 interface HomepageProps {
   timeEntries: Types.TimeEntry[];
 }
 
 export const getServerSideProps = async () => {
-  const response = await getTimeEntries();
-
-  if (response instanceof Error) {
-    return {
-      props: {
-        errorMessage: "This is an error",
-      },
-    };
-  }
+  const { loading, error, data } = await apolloClient.query({
+    query: GET_TIME_ENTRIES,
+  });
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error : {error.message}</p>;
 
   return {
     props: {
-      timeEntries: response,
+      timeEntries: data.allTimeEntries,
     },
   };
 };
 
 const Homepage = ({ timeEntries: initialTimeEntries }: HomepageProps) => {
   const { isModalActive, setIsModalActive } = useContext(StoreContext);
   const { timeEntries, setTimeEntries } = useContext(StoreContext);
-  const [errorWarning, setErrorWarning] = useState("");
+  const [createTimeEntry] = useMutation(ADD_TIME_ENTRY);
+  const [removeTimeEntry] = useMutation(REMOVE_TIME_ENTRY);
 
   useEffect(() => {
     setTimeEntries(initialTimeEntries);
   }, []);
 
-  const handleFormSubmit = async (newTimeEntry: Types.TimeEntry) => {
-    const result = await postTimeEntries(newTimeEntry);
-    if (result instanceof Error) {
-      setErrorWarning("Time entry could not be uploaden");
-      return;
-    }
-    setTimeEntries([...timeEntries, result]);
+  const handleFormSubmit = async ({
+    activity,
+    client,
+    startTimestamp,
+    stopTimestamp,
+  }: Types.TimeEntry) => {
+    const result = await createTimeEntry({
+      variables: {
+        activity,
+        client,
+        startTimestamp,
+        stopTimestamp,
+      },
+    });
+
+    setTimeEntries([...timeEntries, result.data.createTimeEntry]);
     setIsModalActive(false);
   };
 
-  const handleDeleteEntry = async (timeEntryToDelete: Types.TimeEntry) => {
-    if (!window.confirm("are you sure")) {
-      return;
-    }
-    const result = await deleteTimeEntry(timeEntryToDelete);
-    if (result instanceof Error) {
-      setErrorWarning("Time entries could not be deleted");
-      return;
-    }
+  const handleDeleteEntry = async (id: string) => {
+    // console.log(id);
+    await removeTimeEntry({ variables: { id } });
 
-    const afterDelete = timeEntries.filter((entry) => entry.id !== timeEntryToDelete.id);
+    const afterDelete = timeEntries.filter((entry) => entry.id !== id);
```

Comment:
Without spreading you're filtering `timeEntries` here in-place; is this preferred [behavior](https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy)?